### PR TITLE
p4: Check if p4 object is a file and return early if it isn't

### PR DESCRIPTION
**Pull request by EricAtORS**
I tried using p4-fusion on our code base and it was crashing due to associated jobs.

### CODE REVIEW

**Comment by twarit-waikar** [comment]
@EricAtORS Hello! Thank you for sending this PR.

I actually didn't know that `p4 describe` can return results where the `depotFile` key doesn't exist. Can you share some information on in what cases is the `depotFile` key empty?

Based on that we should be able to decide on if that kind of object is worth handling differently or if we can just ignore it.

**Comment by EricAtORS** [comment]
@twarit-waikar 
I haven't had a chance to login to my linux box with the soruce code and the log files, but it's related to Perforce Jobs.

Essentially, we have our Perforce server integrated with Jira and FishEye.
When JIRA creates a new issue in our of projects, it triggers FishEye to generate an associated job. Then, we can tag our changelists with a job id and JIRA/FishEye will update the issue with the associated changelists that we can see from FishEye.

In our case, `varList` was information about that associated job and not any file in our changelist.

**Comment by EricAtORS** [comment]
Also, I'm having issues signing the cla.
I had sent an email to `contactoss-cla@salesforce.com` but I just noticed that the email bounced.

File: p4-fusion/commands/describe_result.cc [hunk 1]
```diff
@@ -13,13 +13,17 @@ void DescribeResult::OutputStat(StrDict* varList)
 int DescribeResult::OutputStatPartial(StrDict* varList)
 {
 	std::string indexString = std::to_string(m_FileData.size());
-	m_FileData.push_back(FileData {});
 
 	StrPtr* depotFile = varList->GetVar(("depotFile" + indexString).c_str());
+	if(depotFile == nullptr){
+		// quick exit if the object returned is not a file
```
Comment (twarit-waikar) [inline p4-fusion/commands/describe_result.cc:19]:
```suggestion
		// Quick exit if the object returned is not a file
```
Minor nitpick :)

File: p4-fusion/commands/describe_result.cc [hunk 2]
```diff
@@ -13,13 +13,17 @@ void DescribeResult::OutputStat(StrDict* varList)
 int DescribeResult::OutputStatPartial(StrDict* varList)
 {
 	std::string indexString = std::to_string(m_FileData.size());
-	m_FileData.push_back(FileData {});
 
 	StrPtr* depotFile = varList->GetVar(("depotFile" + indexString).c_str());
+	if(depotFile == nullptr){
```
Comment (twarit-waikar) [inline p4-fusion/commands/describe_result.cc:18]:
```suggestion
	if(!depotFile)
	{
```
We have been using this style throughout the project, probably should also follow it here

**Review by twarit-waikar — CHANGES_REQUESTED** [review]
Thanks a lot for sending this PR! This was something we didn't realise that could happen.

**Comment by justinharringa** [comment]
> Also, I'm having issues signing the cla. I had sent an email to `contactoss-cla@salesforce.com` but I just noticed that the email bounced.

hey there - have you clicked on the link? Could you help me understand what the issue was? For example, did you click the link? Did you get an error?

Thanks and hope all is well

**Comment by twarit-waikar** [comment]
Hey @EricAtORS, could you check again if you can sign the CLA?

**Comment by twarit-waikar** [comment]
@EricAtORS That should be enough then, I will re-trigger the CLA check on this PR. If you may, you can see my suggestions above, most of them are stylistic differences however

**Comment by twarit-waikar** [comment]
CLA check is completed now. Would like to know if you commit the suggested changes of if I have permission to build on top of your changes so I can make those minor stylistic changes.

Thanks!

**Comment by EricAtORS** [comment]
@twarit-waikar 
Thanks for updating the checks.

You are more than welcome to build on top of my changes to make the stylistic ones. I just haven't had access to my linux bot to do it. If you do it, it would be very appreciated from my part as I'm not sure when I will get access to my linux box.

**Comment by twarit-waikar** [comment]
Thanks for sending the PR!

**Pull request merged**